llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -392,28 +404,30 @@ auto toHazardType(bool Conflict) {
 // recognizing alternatives
 ScheduleHazardRecognizer::HazardType
 AIEHazardRecognizer::getHazardType(const MCInstrDesc &Desc,
-                                   const int DeltaCycles) {
+                                   const int DeltaCycles,
+                                   MemoryBankBits MemoryBanks) {
   return toHazardType(checkConflict(
       Scoreboard, ItinData, Desc.getSchedClass(),
       getSlotSet(Desc, *TII->getFormatInterface(), IgnoreUnknownSlotSets),
-      DeltaCycles, FUDepthLimit));
+      MemoryBanks, DeltaCycles, FUDepthLimit));
 }
 
 ScheduleHazardRecognizer::HazardType
 AIEHazardRecognizer::getHazardType(unsigned SchedClass, SlotBits SlotSet,
-                                   int DeltaCycles) {
+                                   int DeltaCycles, MemoryBankBits MemoryBank) {
   return toHazardType(checkConflict(Scoreboard, ItinData, SchedClass, SlotSet,
-                                    DeltaCycles, FUDepthLimit));
+                                    MemoryBank, DeltaCycles, FUDepthLimit));
 }
 
 bool AIEHazardRecognizer::checkConflict(
     const ResourceScoreboard<FuncUnitWrapper> &Scoreboard,
     const InstrItineraryData *ItinData, unsigned SchedClass, SlotBits SlotSet,
-    int DeltaCycles, std::optional<int> FUDepthLimit) {
+    MemoryBankBits MemoryBank, int DeltaCycles,
+    std::optional<int> FUDepthLimit) {
   assert(Scoreboard.isValidDelta(DeltaCycles));
 
   // Verify format hazards
-  FuncUnitWrapper EmissionCycle(/*Req=*/0, /*Res=*/0, SlotSet);
+  FuncUnitWrapper EmissionCycle(/*Req=*/0, /*Res=*/0, SlotSet, MemoryBank);
```

Comment:
I think this is a bit more complicated. Our memory instructions don't all happen in the same cycle relativ to the emission.  E.g. VST stores in E5, but VST.SRS stores in E7. I think you need to add the MemoryBank bits in `ScoreBoard[DeltaCycles + MemCycle]`. You can find the memory cycle using e.g. `AIEBaseInstrInfo::getFirstMemoryCycle(SchedClass)`

Note you also need to update `getHazardType` in a similar fashion.

Reply:
Thank you for pointing out this, 
Assuming `DeltaCycles` represent E1, so I think I should be adding the MemoryBank bits in `ScoreBoard[DeltaCycles + MemCycle - 1]`